Wrappers/Python/ccpi/viewer/utils/io.py

```diff
@@ -42,3 +66,765 @@ def SaveRenderToPNG(render_window, filename):
+
+        self.logger.info("reading: {}".format(self._FileName))
+
+        progress_callback = kwargs.get('progress_callback')
+
+        reader = self._GetReader(progress_callback)
+        reader.Update()
+        data = reader.GetOutput()
+
+        self._UpdateLoadedImageAttrs(reader, data)
+
+        self._UpdateOriginalImageAttrs(reader)
+
+        return data
+
+    def GetOriginalImageAttrs(self):
+        return self._OriginalImageAttrs
+
+    def GetLoadedImageAttrs(self):
+        return self._LoadedImageAttrs
+
+    def _ValidateRawAttrs(self, raw_image_attrs):
+        if raw_image_attrs is None:
+            return
+        raw_attrs = raw_image_attrs.copy()
+        raw_attrs_schema = Schema({
+            'shape': Or(list, np.ndarray, tuple),
+            'is_fortran': bool,
+            'is_big_endian': bool,
+            'typecode': str
+        })
+        raw_attrs_schema.validate(raw_attrs)
+        return raw_attrs
+
+    def _GetReader(self, progress_callback=None):
+        if os.path.isfile(self._FileName):
+            file_extension = os.path.splitext(self._FileName)[1]
+
+            if file_extension in ['.mha', '.mhd']:
+                reader = self._GetMetaImageReader()
+
+            elif file_extension in ['.npy']:
+                reader = self._GetNumpyImageReader()
+
+            elif file_extension in ['.raw']:
+                reader = self._GetRawImageReader()
+
+            elif file_extension in ['.nxs', '.h5', '.hdf5']:
+                reader = self._GetHDF5ImageReader()
+                self._OriginalImageAttrs['dataset_name'] = self._HDF5DatasetName
+
+            elif file_extension in ['.tif', '.tiff']:
+                image_files = glob.glob(os.path.join(os.path.dirname(self._FileName), '*.{}'.format(file_extension)))
+                if len(image_files) == 0:
+                    raise Exception('No tiff files were found in: {}'.format(self._FileName))
+                self._data = self._GetTiffImageReader(image_files)
+
+            else:
+                raise Exception('File format is not supported. Accepted formats include: .mhd, .mha, .npy, .tif, .raw')
+        else:  # If we are given a folder, not a file, look for tiff files and try to read them
+            image_files = glob.glob(os.path.join(self._FileName, '*.tif')) + glob.glob(
+                os.path.join(self._FileName, '*.tiff'))
+            if len(image_files) == 0:
+                raise Exception('No tiff files were found in: {}'.format(self._FileName))
+            self._data = self._GetTiffImageReader(image_files)
+
+        if file_extension not in ['.tif', '.tiff']:
+            # currently the tiff reader doesn't take these inputs:
+            reader.SetFileName(self._FileName)
+            # if we have acquisition data then we would not resample on
+            # the z axis:
+            reader.SetIsAcquisitionData(not self._ResampleZ)
```

Comment:
This is an interesting learned guess.

Reply:
I updated the comment to explain better about what is happening. Also made this issue https://github.com/vais-ral/CILViewer/issues/309